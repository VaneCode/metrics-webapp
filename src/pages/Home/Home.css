.header {
  width: 100%;
  background-color: #4369b2;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem;
}
.search input {
  color: var(--white-color);
  background-color: transparent;
  border: none;
}
.currenciesList {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
}

.currenciesList a {
  flex: 50%;
}

.currenciesList > a:nth-child(1){
  width: 100%;
}

