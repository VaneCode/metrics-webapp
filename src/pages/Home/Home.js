import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import React, { useState } from 'react';
import styles from './Home.module.css';
import backArrow from '../../img/backArrow.png';
import microphone from '../../img/microphone.png';
import settings from '../../img/settings.png';
import {
  getCurrenciesByName,
  getCurrencyDetails,
} from '../../redux/currencies/currencies';
import Currency from '../../components/Currency/Currency';

const Home = () => {
  const currencyStatus = useSelector((state) => state.currencies);
  console.log(currencyStatus);
  const [filter, setFilter] = useState('');
  const dispatch = useDispatch();
  const handleChange = (p) => {
    const newFilter = {
      id: 1,
      searchWord: p.toLowerCase(),
    };
    setFilter(p);
    dispatch(getCurrenciesByName({ newFilter }));
    console.log(currencyStatus.currenByName);
  };
  let currenList;
  if (!currencyStatus.currenByName.length) {
    currenList = currencyStatus.currencies.map((currency) => (
      <Link
        to="/details"
        key={currency.currencyCode}
        onClick={() => dispatch(getCurrencyDetails({ currency }))}
      >
        <Currency key={currency.currencyCode} currency={currency} />
      </Link>
    ));
  } else {
    currenList = currencyStatus.currenByName.map((currency) => (
      <Link
        to="/details"
        key={currency.currencyCode}
        onClick={() => dispatch(getCurrencyDetails({ currency }))}
      >
        <Currency key={currency.currencyCode} currency={currency} />
      </Link>
    ));
  }
  return (
    <>
      <header className={styles.header}>
        <div>
          <img src={backArrow} alt="home" />
        </div>
        <div className={styles.search}>
          <input
            type="text"
            value={filter}
            placeholder="Currency name"
            onChange={(e) => {
              handleChange(e.target.value);
            }}
          />
        </div>
        <div>
          <img src={microphone} alt="microphone" />
          <img src={settings} alt="settings" />
        </div>
      </header>
      <div className={styles.currenciesList}>{currenList}</div>
    </>
  );
};
export default Home;
