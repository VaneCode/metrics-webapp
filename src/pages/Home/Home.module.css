.header{
  width: 100%;
  background-color: #4369b2;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem;
}

.currenciesList{
  width: 100%;
  display: flex;
  flex-wrap: wrap;
}

.currenciesList div{
  flex: 50%;
}

