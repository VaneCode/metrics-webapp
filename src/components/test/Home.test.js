import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import Home from '../../pages/Home/Home';

describe('Home test', () => {
  test('render', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
