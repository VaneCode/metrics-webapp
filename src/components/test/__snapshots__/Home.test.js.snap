// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test render 1`] = `
Array [
  <header
    className="header"
  >
    <div>
      <img
        alt="home"
        src="backArrow.png"
      />
    </div>
    <div
      className="search"
    >
      <input
        onChange={[Function]}
        placeholder="Currency name"
        type="text"
        value=""
      />
    </div>
    <div>
      <img
        alt="microphone"
        src="microphone.png"
      />
      <img
        alt="settings"
        src="settings.png"
      />
    </div>
  </header>,
  <div
    className="currenciesList"
  />,
]
`;
